nbconvert/preprocessors/extractoutput.py

```diff
@@ -74,6 +79,13 @@ def preprocess_cell(self, cell, resources, cell_index):
         #has been specified
         unique_key = resources.get('unique_key', 'output')
         output_files_dir = resources.get('output_files_dir', None)
+        if self.escape_dir_spaces:
+            unique_key = unique_key.replace(' ', '_')
+            resources['unique_key'] = unique_key
+            if output_files_dir is not None:
+                output_files_dir = output_files_dir.replace(' ', '_')
+                resources['output_files_dir'] = output_files_dir
```

Comment:
I manually tested that ```resources.get('output_files_dir', cwd)``` returns the appropriate directory when called from within ```from_notebook_node```.